lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -1204,7 +1301,8 @@ namespace {
       Space totalSpace = Space::forType(subjectType, Identifier());
       Space coveredSpace = Space::forDisjunct(spaces);
 
-      const size_t totalSpaceSize = totalSpace.getSize(TC, DC);
+      const size_t totalSpaceSize =
+          estimateSearchSpaceSize(totalSpace, coveredSpace);
       if (totalSpaceSize > Space::getMaximumSize()) {
         diagnoseCannotCheck(sawRedundantPattern, totalSpaceSize, coveredSpace,
```

Comment:
You'll have to change the fallback behavior in `diagnoseCannotCheck` too, which tries to guess whether you need a `default`.

Reply:
Alternately, you could pass the old total size to just not change behavior there.

Reply:
Yes, that's what I was thinking, maybe just call `totalSpace.getSize` here and be done, I just haven't looked at what `diagnoseCannotCheck` does yet.

Reply:
Ok, it seems reasonable to re-compute size if estimate was too big, which is exactly what I did, WDYT?